src/sgx/rust_sgx_bridge/rust_sgx_bridge.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ Copyright 2018 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include "poet_enclave_u.h"
+#include "sgx_urts.h"
+#include "rust_sgx_bridge.h"
+#include "poet_enclave.h"
+#include "c11_support.h"
+#include "error.h"
+#include <iostream>
+#include "poet.h"
+
+Poet* Poet::instance = 0;
+
+WaitCertificate* validate_wait_certificate(const char* ser_wait_cert, 
+              const char* ser_wait_cert_sig);
+
+int r_initialize_enclave(r_sgx_enclave_id_t *eid, const char * enclave_path, 
+                         const char *spid) 
+{
+    //check parameters
+    if (!eid || !enclave_path || !spid) {
+        return -1;
+    }
+    try{
+        Poet *poet_enclave_id = Poet::getInstance(enclave_path, spid);
+        //store the enclave id
+        eid->handle = (intptr_t)poet_enclave_id;
+    } catch( sawtooth::poet::PoetError& e) {   
+        return -1;
+    }
+    return 0;
+}
+
+int r_free_enclave(r_sgx_enclave_id_t *eid)
+{
+    if (eid->handle != 0) {
+      Poet *poet_enclave_id = (Poet *)eid->handle;   
+      delete poet_enclave_id;   
+      eid->handle = 0;
+    }
+
+    return 0;
+}
+
+int r_create_signup_info(r_sgx_enclave_id_t *eid, const char *opk_hash, 
```

Comment:
While I do like numbers, instead of using -1 and 0 as return values, please use named error codes defined in enums or constants, or as #defines. for example: const int8_t R_POET_SUCCESS = 0;